### PR TITLE
Fixes #13379 - Delete last_report when switching a host to build mode

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

File: test/unit/host_test.rb [hunk 1]
```diff
@@ -2255,12 +2255,13 @@ def to_managed!
     assert_empty host.fact_values.reload
   end
 
-  test 'reports are deleted when build set to true' do
+  test 'reports and host.last_report are deleted when build set to true' do
     host = FactoryGirl.create(:host, :with_reports)
     assert host.reports.present?
     refute host.build?
+    refute host.last_report.nil?
```
Comment (dLobatog) [inline test/unit/host_test.rb]: I think you mean assert here, this test is failing

File: test/unit/host_test.rb [hunk 2]
```diff
@@ -2255,12 +2255,13 @@ def to_managed!
...
     host.update_attributes(:build => true)
-    assert_empty host.reports.reload
```
Comment (dLobatog) [inline test/unit/host_test.rb:2263]: Why remove this assert? It should still make sure the reports are deleted

**Comment by dLobatog** [comment]
@roidelapluie The test you changed is failing, please take a look :smile: - Can you add the content on your bug description on Redmine to the commit description? Thanks!

File: test/unit/host_test.rb (hunk 2 above)
Comment (roidelapluie) [inline test/unit/host_test.rb:2263]: mistake. I will fix it

File: test/unit/host_test.rb (hunk 1 above)
Comment (roidelapluie) [inline test/unit/host_test.rb]: it looks like  FactoryGirl.create(:host, :with_reports) does not set last_report

File: test/unit/host_test.rb [hunk 3]
```diff
@@ -2255,12 +2255,14 @@ def to_managed!
     assert_empty host.fact_values.reload
   end
 
-  test 'reports are deleted when build set to true' do
+  test 'reports and host.last_report are deleted when build set to true' do
     host = FactoryGirl.create(:host, :with_reports)
     assert host.reports.present?
     refute host.build?
+    refute host.last_report.blank?
```
Comment (dLobatog) [inline test/unit/host_test.rb]: This assertion is failing, there's no `last_report` in the host. I think it'd be nicer if you split this and made two different tests, one that tests the reports are deleted and other that tests last_report is cleared.

Comment (roidelapluie) [inline test/unit/host_test.rb]: I think FactoryGirl.create(:host, :with_reports) should fill last_report. What do you think?

Comment (dLobatog) [inline test/unit/host_test.rb]: That's true too, although I don't think `.create` should be necessary for testing this, `.build_stubbed` should suffice

**Comment by roidelapluie** [comment]
@dLobatog updated

**Comment by roidelapluie** [comment]
@dLobatog Build is green

**Comment by dLobatog** [comment]
Merged as 6d10ed5a5e4b9fe71b847b6488c130baf542f7e1, thanks @roidelapluie!

**Pull request closed without merging**